packages/styled/types/base.d.ts

```diff
@@ -96,6 +146,29 @@ export interface CreateStyledComponent<
  * @example styled('div')<Props>(props => ({ width: props.width })
  */
 export interface CreateStyled {
+  // Component + shouldForwardProp: { include: [] }
```

Comment:
I opted to use overloads here rather than mapped types because i thought it would be simpler